### PR TITLE
Added saturation excess function (for relaxation of q_l and q_i) and cleaned up MoistThermodynamics 

### PR DESCRIPTION
**Pull request by tapios**
The saturation excess function will make it easy to create relaxation sources/sinks for liquid and ice specific humdities, of the form -(q_l - liquid_faction(..) * saturation_excess(..) )/timescale.

### CODE REVIEW

File: src/Utilities/MoistThermodynamics/src/MoistThermodynamics.jl [hunk 1]
```diff
@@ -403,8 +419,8 @@ function saturation_adjustment(e_int, ρ, q_t, T_init = T_triple)
     tol_abs = 1e-3*cv_d
     iter_max = 10
     args = (ρ, q_t, e_int)
-    T0 = max(T_min, air_temperature(e_int, q_t, 0.0, 0.0))
-    T1 = air_temperature(e_int, q_t, 0.0, q_t)
+    T0 = max(T_min, air_temperature(e_int, q_t, eltype(q_t)(0.), eltype(q_t)(0.)))
```
Comment (charleskawczynski) [inline src/Utilities/MoistThermodynamics/src/MoistThermodynamics.jl]:
```suggestion
    T0 = max(T_min, air_temperature(e_int, q_t, typeof(q_t)(0), typeof(q_t)(0)))
```

File: src/Utilities/MoistThermodynamics/src/MoistThermodynamics.jl [hunk 2]
```diff
@@ -403,8 +419,8 @@ function saturation_adjustment(e_int, ρ, q_t, T_init = T_triple)
...
+    T1 = air_temperature(e_int, q_t, eltype(q_t)(0.), q_t)
```
Comment (charleskawczynski) [inline src/Utilities/MoistThermodynamics/src/MoistThermodynamics.jl]:
```suggestion
    T1 = air_temperature(e_int, q_t, typeof(q_t)(0), q_t)
```

**Comment by tapios** [comment]
> I think these changes look great. The only suggestions I have are to change `eltype` (which is for arrays) to `typeof` which is for concrete types. Also, this will promote the type of `0`, so we don't need `0.`.

Thanks, Charlie. Should the other `eltype`'s in MoistThermodynamics than also be changed to `typeof`'s?

**Comment by charleskawczynski** [comment]
Yes, they can be.

**Comment by charleskawczynski** [comment]
Ah, that failure is my fault. I need to change the test case in surface fluxes to not use `rand`. You can change it if you'd like or I can submit a PR to fix that.

**Pull request merged**